arch/risc-v/src/bl602/bl602_spi.c

```diff
@@ -695,10 +709,41 @@ static uint8_t bl602_spi_status(struct spi_dev_s *dev, uint32_t devid)
 static int bl602_spi_cmddata(struct spi_dev_s *dev,
                               uint32_t devid, bool cmd)
 {
+  spiinfo("devid: %" PRIu32 " CMD: %s\n", devid, cmd ? "command" :
+          "data");
+
+#if defined(CONFIG_LCD_ST7735) || defined(CONFIG_LCD_ST7789) || \
+    defined(CONFIG_LCD_GC9A01)
```

Comment:
should we remove #if/#endif and check?

Reply:
Excellent suggestion. I copied this from ESP32-C3, I'm not sure what's the intention of the #if check?

From [esp32c3_board_spi.c](https://github.com/lupyuen/incubator-nuttx/blob/st7789/boards/risc-v/esp32c3/esp32c3-devkit/src/esp32c3_board_spi.c#L58-L86)

```c
#if defined(CONFIG_ESP32C3_SPI2) && defined(CONFIG_SPI_CMDDATA)

int esp32c3_spi2_cmddata(FAR struct spi_dev_s *dev, uint32_t devid, bool cmd)
{
#if defined(CONFIG_LCD_ST7735) || defined(CONFIG_LCD_ST7789) || \
    defined(CONFIG_LCD_GC9A01)
  if (devid == SPIDEV_DISPLAY(0))
    {
      /*  This is the Data/Command control pad which determines whether the
       *  data bits are data or a command.
       */

      esp32c3_gpiowrite(CONFIG_ESP32C3_SPI2_MISOPIN, !cmd);

      return OK;
    }

#endif
  spiinfo("devid: %" PRIu32 " CMD: %s\n", devid, cmd ? "command" :
          "data");

  return -ENODEV;
}

#endif

Reply:
@gustavonihei @Ouss4 do you know the reason?

Reply:
I don't see a reason to specify those LCD devices there, I'd say it's just because it was tested with them.  The check however (`if (devid == SPIDEV_DISPLAY(0))`) needs to stay.

Reply:
Yes, I think so. @lupyuen could you update your patch to more general. Maybe, you can add a new patch to do the same thing for esp32c3 too.

Reply:
Sure @xiaoxiang781216, I'll submit 2 PRs:

1. Remove `#if defined(CONFIG_LCD_ST7735)...` from bl602_spi.c

1. Remove `#if defined(CONFIG_LCD_ST7735)...` from esp32c3_board_spi.c

Thanks :-)

Reply:
The patches have been submitted: https://github.com/apache/incubator-nuttx/pull/5907 and https://github.com/apache/incubator-nuttx/pull/5908. Thank you so much :-)